### PR TITLE
refactor: add typescript [ECO-1027]

### PR DESCRIPTION
**Pull request by karantalapalli**
refactor: add typescript [ECO-1027]

### CODE REVIEW

File: pages/blog/indexPage.vue [hunk 1]
```diff
@@ -51,26 +51,42 @@
...
   components: {
     BlogBanner,
     Devtools,
   },
-  async asyncData(req) {
-    const archivedList = []
-    const recentBlog = []
+  async asyncData(req: any) {
+    const archivedList = [] as any
+    const recentBlog = [] as any
```
Comment (abhishek305) [inline pages/blog/indexPage.vue:84]: define types instead of any

File: pages/blog/indexPage.vue [hunk 2]
```diff
@@ -115,11 +131,11 @@ export default {
   methods: {
     async fetchData() {
       try {
-        const data = await Stack.getEntryByUrl({
+        const data: [Data] = await Stack.getEntryByUrl({
           contentTypeUid: 'page',
           entryUrl: `${this.$route.path}`,
         })
-        const element = document.getElementsByClassName('cslp-tooltip')
+        const element: any = document.getElementsByClassName('cslp-tooltip')
```
Comment (abhishek305) [inline pages/blog/indexPage.vue:138]: define type

File: pages/indexPage.vue [hunk 3]
```diff
@@ -9,17 +9,18 @@
...
 import Stack, { onEntryChange } from '../plugins/contentstack'
+import Req from '@/typescript/pages'
 
 export default {
   components: {
     RenderComponents,
     NotFound
   },
-  async asyncData(req) {
+  async asyncData(req: any) {
```
Comment (abhishek305) [inline pages/indexPage.vue:23]: same here define type

File: pages/indexPage.vue [hunk 4]
```diff
@@ -69,7 +70,7 @@ export default {
           'page_components.section_with_buckets.buckets.description',
         ],
       })
-      const element = document.getElementsByClassName('cslp-tooltip')
+      const element: any = document.getElementsByClassName('cslp-tooltip')
```
Comment (abhishek305) [inline pages/indexPage.vue:73]: same as above comments

File: pages/blog/_blog.vue [hunk 5]
```diff
@@ -34,19 +34,22 @@
...
+import Req from '@/typescript/pages'
+// import PageData from '@/typescript/pages'
 
 export default {
   components: {
     BlogBanner,
     Devtools,
   },
-  async asyncData(req) {
+  async asyncData(req: any) {
```
Comment (abhishek305) [inline pages/blog/_blog.vue:52]: define type

File: pages/blog/_blog.vue [hunk 6]
```diff
@@ -92,13 +95,13 @@ export default {
...
       try {
-        const data = await Stack.getEntryByUrl({
+        const data: [Data] = await Stack.getEntryByUrl({
           contentTypeUid: 'blog_post',
           entryUrl: `${this.$route.fullPath}`,
           referenceFieldPath: [`related_post`, `author`],
           jsonRtePath: ['body', 'related_post.body'],
         })
-        const element = document.getElementsByClassName('cslp-tooltip')
+        const element: any = document.getElementsByClassName('cslp-tooltip')
```
Comment (abhishek305) [inline pages/blog/_blog.vue:104]: here too

**Review by abhishek305 — COMMENTED** [review]
@karantalapalli also update the package.json version currently it's 1.3.0

**Review by Amitkanswal — CHANGES_REQUESTED** [review]
hey @karantalapalli please update package.json version as well

**Review by abhishek305 — APPROVED** [review]

**Review by Amitkanswal — APPROVED** [review]

**Pull request merged**